inception-search-mtas/src/test/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndexTest.java

```diff
@@ -0,0 +1,557 @@
+
+        Waiter createProjectWaiter = new Waiter();
+
+        // Start thread to create and initialize project
+        new Thread(() -> {
+            try {
+                projectService.createProject(project);
+                annotationSchemaService.initializeProject(project);
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+            createProjectWaiter.assertTrue(true);
+            createProjectWaiter.resume();
+        }).start();
+
+        // Wait for thread to complete
+        createProjectWaiter.await();
+
+        SourceDocument sourceDocument = new SourceDocument();
+
+        sourceDocument.setName("test");
+        sourceDocument.setProject(project);
+        sourceDocument.setFormat("text");
+
+        String fileContent = "The capital of Galicia is Santiago de Compostela.";
+
+        InputStream fileStream = new ByteArrayInputStream(
+                fileContent.getBytes(StandardCharsets.UTF_8));
+
+        Waiter uploadWaiter = new Waiter();
+
+        // Start thread to upload the document
+        new Thread(() -> {
+            try {
+                documentService.uploadSourceDocument(fileStream, sourceDocument);
+            }
+            catch (UIMAException | IOException e) {
+                e.printStackTrace();
+            }
+
+            try {
+                // Wait some time so that the document can be indexed
+                Thread.sleep(WAIT_TIME);
+
+                // Test if the index has been created
+                if (searchService.isIndexValid(project)) {
+                    log.info("**************** Index is valid");
+                    uploadWaiter.assertTrue(true);
+                }
+                else {
+                    log.info("**************** Index is NOT valid");
+                    uploadWaiter.assertTrue(false);
+                }
+            }
+            catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            uploadWaiter.resume();
+        }).start();
+
+        // Wait for thread to complete
+        uploadWaiter.await();
+
+        User user = userRepository.get("admin");
+
+        String query = "Galicia";
+
+        List<SearchResult> results = null;
+
+        // Execute query
+        results = (ArrayList<SearchResult>) searchService.query(user, project, query);
+
+        // Test results
+        SearchResult expectedResult = new SearchResult();
+        expectedResult.setDocumentId(sourceDocument.getId());
+        expectedResult.setDocumentTitle("test");
+        expectedResult.setText("Galicia ");
+        expectedResult.setLeftContext("capital of ");
+        expectedResult.setRightContext("is ");
+        expectedResult.setOffsetStart(15);
+        expectedResult.setOffsetEnd(22);
+        expectedResult.setTokenStart(3);
+        expectedResult.setTokenLength(1);
+
+        ArrayList<SearchResult> expectedResults = new ArrayList<SearchResult>();
```

Comment:
`List = ArrayList`

Reply:
You do not need `expectedResults `, do you?

Reply:
It was a leftover, I've removed it.